File: 프로그래머스/K번째수/knum.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
	vector<int> answer;
	for (int i = 0; i < commands.size(); i++) {
		int start = commands[i][0]-1;
		int end = commands[i][1];
		int selectNum = commands[i][2];
		vector<int> curarray;
		for (int j = start; j < end; j++) {
			curarray.push_back(array[j]);
		}
		sort(curarray.begin(), curarray.end());
		answer.push_back(curarray[selectNum - 1]);
	}
	return answer;
}

File: 프로그래머스/K번째수/knum.java
package com.company;
import java.util.Arrays;
import java.util.ArrayList;
import java.util.List;
class Solution {
    public int[] solution(int[] array, int[][] commands) {
        ArrayList<Integer> arr = new ArrayList<>();
        for(int i=0; i<commands.length;i++){
            int[] temparry = Arrays.copyOfRange(array,commands[i][0]-1,commands[i][1]);
            Arrays.sort(temparry);
            arr.add(temparry[commands[i][2]-1]);
        }
        int []answer = arr.stream().mapToInt(Integer::intValue).toArray();
        return answer;
    }
}
public class Main {
    public static void main(String[] args) {
        Solution a = new Solution();

    }
}
